Creates a completely new website with multiple pages

laura-barluzzi opened the pull request:
This PR entirely refactores the existing website of www.ascoderu.ca

## Why
This PR aims to improve Ascoderu's **online presence** by:
1. Adding more information about the organization
2. Adding more information about Lokole as a device and product
3. Adding more information about the history and the team involved in the effort
4. Adding links to media coverage, articles etc.
5. Adding actions that people can take to help Ascoderu

## How
This version is built entirely using **HTML**, **CSS** and **Javascript.** However,  the following libraries are applied: _Bootstrap_, _Bootstrap Dialog_, & _Fullpage.js_
This version should work fine on most devices (including smart-phone, tablets and touch devices).

c-w commented:
This is great, thanks so much for the PR. I'll split the review into multiple parts for making it easier, first focusing on functionality and later on code.

### Inconsistency: no way to return to the home screen

![image](https://user-images.githubusercontent.com/1086421/34653160-f0325cb6-f3b5-11e7-9a9e-c119e6f6cd3d.png)

It would be good to have a menu item called "Home" that links back to the first page. This removes two inconsistencies, namely that there is no menu item highlighted on first load of the page and that two different ways to navigate the sections are required (scroll for first section, menu+scroll for all others). Removing the inconsistencies will decrease confusion potential for the users.

### Avoid loading too large images

![image](https://user-images.githubusercontent.com/1086421/34653182-9183d9aa-f3b6-11e7-9ca9-106062b1c9ed.png)

All the images in the site should be resized to the correct size that is being displayed in the page. For example, for the image in the screenshot above we're loading a 3844x2910 image although we're only displaying it at 290x220. Other images have the same issue: e.g. team.jpg, laura.jpg, logo-1200x630.png, etc. Resizing the images will fix reduce the bandwidth required for the page to load (important on mobile connections) and will also decrease the chance that we have a flash of a white placeholder while the image is loading.

### Fix wording

![image](https://user-images.githubusercontent.com/1086421/34653238-45e1342e-f3b7-11e7-8714-f28bbd830d77.png)

For consistency with the other items, `<code/>` should just be `code`.

![image](https://user-images.githubusercontent.com/1086421/34653252-6cc28390-f3b7-11e7-941f-125b62b58fe3.png)

The term "remote email c" should be "remote email center".

![image](https://user-images.githubusercontent.com/1086421/34653276-b5d1ee72-f3b7-11e7-9005-60f96acb462f.png)

Should this be something like "like" or "following the"?

c-w commented on index.html:
Missing opening quotation mark at the start of the href. `href='https://face2faceafrica.com/...`

c-w commented on index.html:
The ampersand (&) is a special character in html (used to introduce escape sequences for special characters). For displaying a literal ampersand, use `&amp;`. [More information](https://stackoverflow.com/a/3493425/3817588).

c-w commented on script.js:
Instead of having hard to maintain long strings in the javascript, why not used a hidden div in the html to specify these strings? Keeping content in html and logic in javascript is easier to maintain since someone wouldn't have to check multiple places (html and js) to just update the content. Having all the content in one place also makes future changes like localization easier.

You could do something like this, similar to what's happening with the modals setup code on lines 63-71 below:

```js
var $info1 = $('#info1');
var info1ContentId = $info1.data('content-reference');
var $info1Content = $('#' + info1ContentId);
var info1Content = $info1Content.text();
$info1.data('content', info1Content);
```

laura-barluzzi replied:
Actually this is the old version of what was already moved to hidden divs. I just forgot to delete it. I'm going to delete this chunk since its functionality is already covered by the hidden divs and this function in `script.js`
```
  $("div[rel='modal']").click(function(event){
    var clickedModalId = $(this).data("reference");
    var contentModalHtml = $("#"+clickedModalId).html();

    BootstrapDialog.show({
      title: $(this).data("title"),
      message: $('<div></div>').append(contentModalHtml)
    });
  });
```

c-w commented on script.js:
Why wrap this in a function? That's unnecessary here. Normally we wrap code in functions so that variables don't leak out of the scope. For example:

```
(function() {
  var foo = '1';
  console.log(foo); // foo is defined, prints 1
})();
console.log(foo); // foo is not defined
```

However, here we don't have any variables that would leak, so might as well remove the function.

c-w commented on script.js:
Rule of thumb: comments usually indicate that you want to have a named function instead.

c-w commented on script.js:
What does this block do? Please extract a named function to make it clearer.

c-w commented on script.js:
```diff
@@ -0,0 +1,73 @@
+function setBackgroundColor(heightScreen) {
+  if (heightScreen > 1000) {
+    $("#home-page").css("background-image", "none");
+    $("#home-page").css("background-color", "rgb(133, 57, 98)");
```
Note: you can pass an object to the jquery css function to update multiple styles at the same time with a single call ([docs](http://api.jquery.com/css/#css-properties)), e.g.:

```js
$('#home-page').css({
  'backgroud-image': 'none',
  'background-color': 'rgb(133, 57, 98)'
});
```

c-w commented on script.js:
What does this function actually do? Seems to return a height and width? Could the name be made more explicit to better explain what this does?

c-w commented on script.js:
Note: here you're returning an array and the caller has to know exactly what the array represents, here: height and then width. You can remove this implicit knowledge and make it explicit by returning an object instead:

```js
var small = {height: 800, width: 1000};
var heightIsLower = height < small.height;
var widthIsLower = width < small.width;
return heightIsLower && widthIsLower ? small : {height: 0, width: 0};
```

laura-barluzzi replied:
Since I removed `setBackgroundColor(heightScreen);` this function does not have a purpose anymore. `checkDevice` was created having in mind the need to check the device screen dimension in order to set a background image or color accordingly. This functionality is no longer required.

c-w commented on script.js:
```diff
@@ -0,0 +1,73 @@
+function setBackgroundColor(heightScreen) {
+  if (heightScreen > 1000) {
+    $("#home-page").css("background-image", "none");
```
Generally, it is preferred to keep all styling information in the css file so that we only have to check one place to know all the styles an application is using and to avoid having to comb through the logic in the js file to update styling information in the future. As such, it would be more maintainable here to assign a css class instead of the inline styles.

```css
#home-page.largeHeightScreen {
  backgroud-image: none;
  backgroud-color: rgb(133, 57, 98);
}
```

```js
if (heightScreen > 1000) {
  $('#home-page').addClass('largeHeightScreen');
}
```

laura-barluzzi replied:
I remove this feature in full since it turned out that the background image works fine for all heights.

c-w commented on script.js:
Nit: this can just be `$('<div/>')`

c-w commented on script.js:
Similar comment to the `#home-page` one above. You can just use a css class here. Specifically, you can use bootstrap's `.hidden` class in the html and remove it here.

laura-barluzzi replied:
if I add `class="hidden"` in the HTML and then fire `$("#website-content").removeClass("hidden");` in the script.js it doesn't work. The "hidden" class doesn't get removed.

I went for a slightly different approach; in css I set `#website-content {display: none}` and then I in afterRender I fire `$("#website-content").show();`

In this way I obtain the same. I am not sure why the first version didn't work out though.

c-w commented on style.css:
```diff
@@ -1,80 +1,284 @@
 
-.page-container {
+#website-content {
+  display: none;
```
This can be replaced by adding the bootstrap utility class `hidden` in the html. Same for `modal-1-content` below.

c-w commented on index.html:
Typography nitpick: technically a em-dash `&mdash;` would be more appropriate here than a hyphen. [More information](http://www.thepunctuationguide.com/em-dash.html).

laura-barluzzi commented:
Thanks @c-w for the review. I'll address these tomorrow. I just wanted to point out that we previously discussed about the linking to the home-page from the nav-bar and we both agreed that it was fine to use the upper-left Ascoderu logo as a home-page link. Indeed, if you click there you land to the home-page again.

So the options are two:
1. I do what you suggested and I remove the link from the logo. In this way the logo remains but is for "beauty purpose only".
2. I keep the current version with the logo being a button linking to the home-page.

Which one should I do in your opinion?

laura-barluzzi commented on style.css:
```diff
@@ -1,80 +1,284 @@
-  max-width: 30%;
-  text-align: center;
+  justify-content: flex-start;
+  align-items: start;
```
Yes. But apparently it works even if you just use "start". See [this tutorial](https://www.w3schools.com/cssref/tryit.asp?filename=trycss3_align-items).

I'll change it to flex-start to avoid problems though.

laura-barluzzi commented:
@c-w I addressed all your comments. There are only a couple of things in `script.js` that I haven't done since I realized they were useless or meant-to-be-deleted old pieces of code.

Let me now if you find something else before merging the PR.
